pkg/telemetry/init.go

```diff
@@ -2,63 +2,42 @@ package telemetry
 
 import (
 	"context"
-	"fmt"
 
-	gcfg "go.minekube.com/gate/pkg/gate/config"
+	"go.minekube.com/gate/pkg/gate/config"
 )
 
-// Init initializes OpenTelemetry with the configured exporters and providers
-func Init(ctx context.Context, cfg *gcfg.Config) (cleanup func(), err error) {
-	if cfg == nil {
-		return nil, fmt.Errorf("config is nil")
-	}
+// Init initializes OpenTelemetry with configuration from environment variables and config.
+// It returns a cleanup function and any error encountered.
+func Init(ctx context.Context, cfg *config.Config) (func(), error) {
+	// Apply default telemetry config first
+	cfg = WithDefaults(cfg)
 
-	// Initialize telemetry with config
-	if cfg.Telemetry.Metrics.Enabled || cfg.Telemetry.Tracing.Enabled {
-		cleanup, err = initTelemetry(ctx, cfg)
-		if err != nil {
-			return nil, fmt.Errorf("failed to initialize telemetry: %w", err)
-		}
-		return cleanup, nil
+	// Create new telemetry instance
+	_, cleanup, err := New(ctx, cfg)
+	if err != nil {
+		return nil, err
 	}
-
-	return func() {}, nil // Return no-op cleanup if telemetry disabled
+	return cleanup, nil
 }
 
-// WithDefaults returns a config with telemetry enabled using reasonable defaults
-func WithDefaults(cfg *gcfg.Config) *gcfg.Config {
+// WithDefaults returns a copy of the config with default telemetry settings applied.
+func WithDefaults(cfg *config.Config) *config.Config {
```

Comment:
`func WithDefaults` should be removed as we use the other default strategy and have validation